kernel/kernel-api/src/main/java/io/delta/kernel/Table.java

```diff
@@ -66,4 +66,15 @@ Snapshot getLatestSnapshot(TableClient tableClient)
      */
     Snapshot getSnapshotAtVersion(TableClient tableClient, long versionId)
         throws TableNotFoundException;
+
+    /**
+     * Get the snapshot at the given {@code timestamp}.
+     *
+     * @param tableClient {@link TableClient} instance to use in Delta Kernel.
+     * @param millisSinceEpochUTC timestamp to fetch the snapshot for in milliseconds since the
+     *                            unix epoch
+     * @return an instance of {@link Snapshot}
+     */
+    Snapshot getSnapshotAtTimestamp(TableClient tableClient, long millisSinceEpochUTC)
```

Comment:
We will eventually need 2 types of time travel semantics for batch and streaming semantics

- getVersionBeforeOrAtTimestamp
- getVersionAtOrAfterTimestamp

We should think a bit more about this public API (method name) ... so that adding the other method doesn't cause confusion

Reply:
Standalone `DeltaLog`, the above-mentioned methods return a version instead of a snapshot. 

Looking at Flink how these APIs are used:
* can't find `getVersionBeforeOrAtTimestamp` used anywhere.
* `getVersionAtOrAfterTimestamp` is used as to get the version immediately followed by `getSnapshotAtVersion(version returned by getVersionAtOrAfterTimestamp).

what if we provide two APIs:
`getSnapshotBeforeOrAtTimestamp`
`getSnapshotAtOrAfterTimestamp`

Are there any case where we just need the version?

Reply:
@vkorukanti was there an ask from delta-sharing to support version getting in this scenario? Or is just the snapshot sufficient

Reply:
We can rename this method to `getSnapshotBeforeOrAtTimestamp`? I'm wondering if that's a bit hard to parse for the simple batch time-travel case

Reply:
Looking at the [Delta sharing protocol](https://github.com/delta-io/delta-sharing/blob/main/PROTOCOL.md#query-table-version), it seems like sharing needs `atOrAfter`. @linzhou-db Please confirm.

Reply:
It also seems like this requires `versionAtOrAfter` to get the version (to avoid unnecessarily loading the snapshot metadata/protocol)

Reply:
yes, delta sharing do need both snapshot and version AtOrAfter a timestamp.

Reply:
Created https://github.com/delta-io/delta/issues/2679 to revisit this API + add additional functionality